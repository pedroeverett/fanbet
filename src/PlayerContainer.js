import React from 'react';
import styled from 'styled-components';
import { Button } from 'semantic-ui-react';

const PlayerWraper = styled('div')`
  text-align: center;
  display: inline-block;
  width: 200px;
`;

const PlayerName = styled('h4')`
  padding-bottom: 20px;
`;

const Image = styled('img')`
  height: 200px;
  width: 200px;
`;

const PlayerContainer = ({ player, onClick }) => {
  return (
    <PlayerWraper>
        <div>
          <Image src={player.images.default.url} alt={player.first_name} />
          <PlayerName>Name: {player.first_name} {player.last_name}</PlayerName>
        </div>
        <Button onClick={onClick}>Choose Player</Button>
    </PlayerWraper>
  );
};

export default PlayerContainer;