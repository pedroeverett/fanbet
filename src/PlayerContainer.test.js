import React from 'react';
import { shallow } from 'enzyme';
import PlayerContainer from './PlayerContainer';

describe('PlayerContainer', () => {
    const playerMoch = {"first_name": "Stephen",
        "fixture": {
            "_members": ["112160"],
            "_ref": "fixtures.id"
        },
        "fppg": 47.94303797468354,
        "id": "15475-9524",
        "images": {
            "default": {
                "height": 200,
                "url": "https://d17odppiik753x.cloudfront.net/playerimages/nba/9524.png",
                "width": 200
            }
        },
        "injured": false,
        "injury_details": "knee",
        "injury_status": "o",
        "last_name": "Curry",
        "news": {
            "latest": "2016-05-02T18:35:15Z"
        },
        "played": 79,
        "player_card_url": "https://www.fanduel.com/e/Player/9524/Stats/15475",
        "position": "PG",
        "removed": false,
        "salary": 10600,
        "starting_order": null,
        "team": {
            "_members": ["687"],
            "_ref": "teams.id"
        }
    }
    const component = shallow(<PlayerContainer player={playerMoch} />);
    it('should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

//   it('should get props correctly', () => {
//     expect(player.first_name).toBe('pedro');
//   });

//   it('should contain 1 div element', () => {
//     expect(component.find('div').length).toBe(1);
//   })

});
