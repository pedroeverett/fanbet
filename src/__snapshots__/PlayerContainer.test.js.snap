// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerContainer should render correctly 1`] = `
<styled.div>
  <div>
    <styled.img
      alt="Stephen"
      src="https://d17odppiik753x.cloudfront.net/playerimages/nba/9524.png"
    />
    <styled.h4>
      Name: 
      Stephen
       
      Curry
    </styled.h4>
  </div>
  <Button
    as="button"
  >
    Choose Player
  </Button>
</styled.div>
`;
